src/models/transaction/stellar_types.rs

```diff
@@ -0,0 +1,594 @@
+                let text = stellar_xdr::curr::StringM::<28>::try_from(s.as_str()).map_err(|e| {
+                    SignerError::ConversionError(format!("Invalid memo text: {}", e))
+                })?;
+                Memo::Text(text)
+            }
+            MemoSpec::Id(i) => Memo::Id(i),
+            MemoSpec::Hash(b) => Memo::Hash(Hash(b)),
+            MemoSpec::Return(b) => Memo::Return(Hash(b)),
+        })
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum AssetSpec {
+    Native,
+    Credit4 { code: String, issuer: String },
+    Credit12 { code: String, issuer: String },
+}
+
+impl TryFrom<AssetSpec> for Asset {
+    type Error = SignerError;
+    fn try_from(a: AssetSpec) -> Result<Self, Self::Error> {
+        Ok(match a {
+            AssetSpec::Native => Asset::Native,
+            AssetSpec::Credit4 { code, issuer } => {
+                let b = code.as_bytes();
+                if !(1..=4).contains(&b.len()) {
+                    return Err(SignerError::ConversionError("asset code 1-4 chars".into()));
+                }
+                let mut buf = [0u8; 4];
+                buf[..b.len()].copy_from_slice(b);
+
+                let issuer_pk = PublicKey::from_str(&issuer)
+                    .map_err(|e| SignerError::ConversionError(format!("Invalid issuer: {}", e)))?;
+
+                let uint256 = stellar_xdr::curr::Uint256(issuer_pk.0);
+                let pk = XdrPublicKey::PublicKeyTypeEd25519(uint256);
+                let account_id = AccountId(pk);
+
+                Asset::CreditAlphanum4(AlphaNum4 {
+                    asset_code: AssetCode4(buf),
+                    issuer: account_id,
+                })
+            }
+            AssetSpec::Credit12 { code, issuer } => {
+                let b = code.as_bytes();
+                if !(5..=12).contains(&b.len()) {
+                    return Err(SignerError::ConversionError("asset code 5-12 chars".into()));
+                }
+                let mut buf = [0u8; 12];
+                buf[..b.len()].copy_from_slice(b);
+
+                let issuer_pk = PublicKey::from_str(&issuer)
+                    .map_err(|e| SignerError::ConversionError(format!("Invalid issuer: {}", e)))?;
+
+                let uint256 = stellar_xdr::curr::Uint256(issuer_pk.0);
+                let pk = XdrPublicKey::PublicKeyTypeEd25519(uint256);
+                let account_id = AccountId(pk);
+
+                Asset::CreditAlphanum12(AlphaNum12 {
+                    asset_code: AssetCode12(buf),
+                    issuer: account_id,
+                })
+            }
+        })
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ToSchema)]
+#[serde(tag = "op", rename_all = "snake_case")]
+pub enum OperationSpec {
+    Payment {
+        destination: String,
+        amount: i64,
+        asset: AssetSpec,
+    },
+}
+
+impl TryFrom<OperationSpec> for Operation {
```

Comment:
We are decoding xdr transaction data in order to have it for internal processing?

Do we need it for validations or fee estimations?

Mainly asking do we need it and do we need it for all fields or whether to keep it in serialized format.

Reply:
We are decoding for signing time and for simulations etc. We do need all fields (and eventually there could be more as we add more op types).